File: 3_2_winform_protocol_structure_demo/3_2_winform_protocol_structure_demo/Protocol/Work.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HYKT
{
    class Work
    {
        public Work() {
            FailedTimes = 0;
        }

        public int FailedTimes { get; set; }
        public byte[] Data { get; set; }

        public bool CheckResponse(byte[] returnData) {
            if (returnData[0] == Data[2] && returnData[1] == Data[3]) { return true; }
            return false;
        }

    }
}

File: 13_3_stm32_iap_downloader_tool/src/protocol/work.cpp


#include "work.h"

#include "../utils/byteutil.h"


Work::Work()
{
    failed_times = 0;
}

bool Work::CheckResponse(quint8 *return_pkg) {

    quint8 src_pkg_bit_1 = ByteUtil::Char2Byte(data.at(2));//command type
    quint8 src_pkg_bit_2 = ByteUtil::Char2Byte(data.at(3));//command code

   if (return_pkg[0] == src_pkg_bit_1 && return_pkg[1] == src_pkg_bit_2) {
       return true;
   } else {
       return false;
   }
}



